x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/utils/bulk_create_suppressed_alerts_in_memory.ts

```diff
@@ -99,6 +136,105 @@ export const bulkCreateSuppressedAlertsInMemory = async ({
   return executeBulkCreateAlerts({
     suppressibleWrappedDocs,
     unsuppressibleWrappedDocs,
+    buildingBlockAlerts,
+    toReturn,
+    bulkCreate,
+    services,
+    ruleExecutionLogger,
+    tuple,
+    alertSuppression,
+    alertWithSuppression,
+    alertTimestampOverride,
+    experimentalFeatures,
+    maxNumberOfAlertsMultiplier,
+  });
+};
+
+/**
+ * wraps, bulk create and suppress alerts in memory, also takes care of missing fields logic.
+ * If parameter alertSuppression.missingFieldsStrategy configured not to be suppressed,
+ * regular alerts will be created for such events without suppression
+ */
+export const bulkCreateSuppressedSequencesInMemory = async ({
+  sequences,
+  toReturn,
+  wrapSequences,
+  bulkCreate,
+  services,
+  buildReasonMessage,
+  ruleExecutionLogger,
+  tuple,
+  alertSuppression,
+  wrapSuppressedSequences,
+  alertGroupFromSequenceBuilder,
+  addSequenceSuppressionTermsAndFields,
+  alertWithSuppression,
+  alertTimestampOverride,
+  experimentalFeatures,
+  mergeSourceAndFields = false,
+  maxNumberOfAlertsMultiplier,
+}: BulkCreateSuppressedSequencesParams) => {
+  const suppressOnMissingFields =
+    (alertSuppression?.missingFieldsStrategy ?? DEFAULT_SUPPRESSION_MISSING_FIELDS_STRATEGY) ===
+    AlertSuppressionMissingFieldsStrategyEnum.suppress;
+
+  const suppressibleWrappedSequences: Array<
+    WrappedFieldsLatest<EqlShellFieldsLatest & SuppressionFieldsLatest> & {
+      subAlerts: Array<WrappedFieldsLatest<EqlBuildingBlockFieldsLatest>>;
+    }
+  > = [];
+  const unsuppressibleWrappedDocs: Array<WrappedFieldsLatest<BaseFieldsLatest>> = [];
+
+  sequences.forEach((sequence) => {
+    const alertGroupFromSequence = alertGroupFromSequenceBuilder(sequence, buildReasonMessage);
+    const shellAlert = alertGroupFromSequence?.[0];
+    if (shellAlert != null) {
+      if (!suppressOnMissingFields) {
+        const [shellAlertWithFields] = partitionMissingFieldsEvents(
```

Comment:
as discussed on zoom, it should be easier to extract the logic for determining if an event has the necessary fields and apply it to the `shellAlertWithFields` directly rather than coercing a single event into the form necessary to reuse `partitionMissingFieldsEvents`

Reply:
Fixed here: https://github.com/elastic/kibana/pull/189725/commits/0412e2593ae4c596625344ff688d7112cf747356#diff-7bc3e0677cda5acc5376a472970bd2a6da57a71df78cdc0c77f1d4ce3182410aR188-R190